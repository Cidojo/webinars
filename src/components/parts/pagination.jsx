import * as React from 'react';
import {Link} from 'react-router-dom';

const Pagination = (props) => {
  const {pages, currentPage} = props;

  return (
    <div className='pagination'>
      <ul>
        {[...Array(pages)].map((page, i) => (
          <li
            key={i}
            className={currentPage === i + 1 ? `pagination__item pagination__item--active` : 'pagination__item'}
          >
            <Link to={`/?page=${i + 1}`}>{i + 1}</Link>
          </li>
        ))}
      </ul>
    </div>
  );
};

export {Pagination};
