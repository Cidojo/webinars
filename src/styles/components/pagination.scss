.pagination {
  margin-bottom: 30px;

  ul {
    @extend %reset-list;
    @extend %container;

    display: flex;
    justify-content: center;
    padding-top: 10px;
    padding-bottom: 10px;
  }
}

.pagination__item {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 34px;
  height: 34px;
  margin: 6px;
  color: $color-primary-blue;
  background-color: $color-brighter-grey;

  &--active {
    color: #fff;
    background-color: $color-primary-blue;
  }
}
