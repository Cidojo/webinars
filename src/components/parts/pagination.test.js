import * as React from 'react';
import * as renderer from 'react-test-renderer';
import {BrowserRouter} from 'react-router-dom';
import {Pagination} from './pagination.jsx';

const props = {
  pages: 0,
  currentPage: 0
};

it(`should render Pagination component without errors`, () => {
  const tree = renderer.create(<BrowserRouter><Pagination {...props} /></BrowserRouter>).toJSON();
  expect(tree).toMatchSnapshot();
});
