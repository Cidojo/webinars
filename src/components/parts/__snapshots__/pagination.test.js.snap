// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Pagination component without errors 1`] = `
<div
  className="pagination"
>
  <ul />
</div>
`;
